echo_server_single_window.py

```diff
@@ -0,0 +1,41 @@
+import socket
+import thread
+import echo_client
+import time
+buffersize = 8
+
+
+def start_server():
+    server_socket = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_IP
+        )
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(1)
+    try:
+        while True:
+            time.sleep(1)
+            msg = raw_input("Enter statement to echo: ")
+            thread.start_new_thread(wait_then_start_client, (msg,))
```

Comment:
this allows you to run the client while the server is running.  Clever, But it doesn't solve the problem of testing the client without starting the server separately, does it?